### PR TITLE
Release 4.14 azure

### PR DESCRIPTION
**Pull request by bitoku**
Followed https://github.com/Azure/ARO-RP/blob/master/docs/upstream-differences.md

### CODE REVIEW

**Comment by bitoku** [comment]
Included https://github.com/jewzaam/installer-aro/commit/1ad7d31f4d6f1c53e246a1e51b17efd74174d4b2 to https://github.com/jewzaam/installer-aro/pull/29/commits/641c2405522aa0d4f267804b5bc0639e96783f19

File: data/data/manifests/bootkube/aro-ingress-service.yaml.template [hunk 1]
```diff
@@ -0,0 +1,25 @@
+apiVersion: v1
```
Comment (rolandmkunkel) [inline data/data/manifests/bootkube/aro-ingress-service.yaml.template:1]: This used to be in a separate commit which is not present anymore. What is the reason for that?
https://github.com/jewzaam/installer-aro/commit/7f24b4ac63fff36be31a6335b390333a3fc34a8f

File: data/data/manifests/bootkube/aro-imageregistry.yaml.template [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: imageregistry.operator.openshift.io/v1
```
Comment (rolandmkunkel) [inline data/data/manifests/bootkube/aro-imageregistry.yaml.template:1]: This used to be in commit HACK: mark image registry storage unmanaged 
https://github.com/jewzaam/installer-aro/commit/d6e53ba050e710d9dc221012686b69bd83c539fe

**Review by rolandmkunkel — COMMENTED** [review]
I think the openshift installer of 4.14 also bumped the minimum golang to 1.20 https://github.com/openshift/installer/blob/release-4.14/hack/build.sh

**Comment by bitoku** [comment]
I had some commit merged by mistake when I rebased it.
I'll redo it tomorrow.

**Comment by bitoku** [comment]
> I think the openshift installer of 4.14 also bumped the minimum golang to 1.20

Do you think we need this kind of change to this branch? https://github.com/jewzaam/installer-aro/commit/dc5bd1fc0ad195722d8ed73864045d18664c60de

In that case, this branch is based on the openshift/installer:release-4.14, so bumping commits are already included.

https://github.com/bitoku/installer/blob/my-release-4.14-azure/go.mod#L3
https://github.com/bitoku/installer/blob/my-release-4.14-azure/hack/build.sh#L29
https://github.com/bitoku/installer/blob/my-release-4.14-azure/hack/go-test.sh#L11

**Review by rolandmkunkel — APPROVED** [review]
